parlai/scripts/train_model.py

```diff
@@ -630,8 +653,56 @@ def _compute_eta(self, epochs_completed, time_elapsed):
             if eta is None or time_left < eta:
                 eta = time_left
 
+        max_train_steps = self.opt.get('max_train_steps', -1)
+        if max_train_steps > 0:
+            steps_progress = steps_taken / max_train_steps
+            eta = (1 - steps_progress) * time_elapsed / steps_progress
+
         return eta
 
+    def _get_time(self, world: World) -> Tuple[float, float, float]:
+        """
+        Return train, log, and validate timing.
+
+        If relying on the time for validation/logging/max train time purposes,
+        we sync and return primary worker's time.
+
+        Otherwise, it's not super relevant what we do here.
+
+        **SIDE EFFECT**: Update _total_epochs trained.
+
+        :param world:
+            current running world
+
+        :return (train, log, valid):
+            return time for each of train, log, and validation
+        """
+        if any(
+            getattr(self, k) < float('inf')
```

Comment:
Can we just expand this? I know it's more verbose

Reply:
sure